src/transformers/modeling_tf_utils.py

```diff
@@ -351,6 +353,15 @@ def input_processing(func, config, input_ids, **kwargs):
         )
         output["past_key_values"] = kwargs["kwargs_call"].pop("decoder_cached_states")
 
+    # `TFGPT2` has `past` parameter while other models use `past_key_values`
+    # It is probably better to change `TFGPT2`'s parameter instead of adding this block.
+    # (The past keys/values are added only if the model accepts it - otherwise we should leave it in `kwargs_call`)
+    if "past" in kwargs["kwargs_call"] and "past_key_values" in kwargs:
```

Comment:
oh yeah we forgot to correctly change `past` to `past_key_values` in TFGPT2 it seems! However now changing `past` to `past_key_values` is quite a breaking change...not sure it's worth it cc @LysandreJik @sgugger